engine/record_tx_strategy_outgoing_tx.go

```diff
@@ -121,23 +136,33 @@ func _hydrateOutgoingWithDraft(ctx context.Context, tx *Transaction) error {
 	return nil // success
 }
 
-func _hydrateOutgoingWithSync(tx *Transaction) {
-	sync := newSyncTransaction(tx.ID, tx.draftTransaction.Configuration.Sync, tx.GetOptions(true)...)
-
-	// setup synchronization
-	sync.SyncStatus = SyncStatusPending // wait until transaction is broadcasted or P2P provider is notified
+func _shouldNotifyP2P(tx *Transaction) bool {
+	return slices.ContainsFunc(tx.draftTransaction.Configuration.Outputs, func(o *TransactionOutput) bool {
+		return o.PaymailP4 != nil && o.PaymailP4.ResolutionType == ResolutionTypeP2P
+	})
+}
 
-	sync.Metadata = tx.Metadata
+func _handleNotifyP2PError(ctx context.Context, c ClientInterface, transaction *Transaction) error {
+	logger := c.Logger()
+	chainstateService := c.Chainstate()
 
-	sync.transaction = tx
-	tx.syncTransaction = sync
-}
+	saveAsProblematic := func() {
+		transaction.TxStatus = TxStatusProblematic
+		if err := transaction.Save(ctx); err != nil {
+			logger.Error().Str("txID", transaction.ID).Err(err).Msg("Error saving transaction after notifyP2P failed")
+		}
+	}
 
-func _shouldNotifyP2P(tx *Transaction) bool {
-	for _, o := range tx.draftTransaction.Configuration.Outputs {
-		if o.PaymailP4 != nil && o.PaymailP4.ResolutionType == ResolutionTypeP2P {
-			return true
+	_, err := chainstateService.QueryTransaction(ctx, transaction.ID, chainstate.RequiredInMempool, defaultQueryTxTimeout)
+	if errors.Is(err, spverrors.ErrCouldNotFindTransaction) {
+		if err := c.RevertTransaction(ctx, transaction.ID); err != nil {
+			saveAsProblematic()
+			return spverrors.ErrProcessP2PTx.Wrap(err)
 		}
+
+		// RevertTransaction saves the transaction itself as REVERTED
+		return spverrors.ErrProcessP2PTx
 	}
-	return false
+	saveAsProblematic()
```

Comment:
the function is used twice here. 
- Check the entire function `_handleNotifyP2PError`
- Consider the conditions
- If you have more readable solution - please help me

Reply:
ok, at this stage of `engine` package I don't see any good (or better) way of doing this,
but as you see, how it looks is a little bit not readable because I didn't saw that this function was called twice

Reply:
⚠️  Or maybe it's better to just add this `SaveAsProblematic` to transaction model

Reply:
`SaveAsProblematic` in transaction model would not solve the problem and purpose of this local function.
- if I did as you suggested - I would have to put the logging line twice.